src/main/java/org/jenkinsci/plugins/workflow/support/concurrent/Futures.java

```diff
@@ -324,7 +324,7 @@ public void run() {
     public static <V> ListenableFuture<List<V>> allAsList(
         ListenableFuture<? extends V>... futures) {
       return new ListFuture<V>(ImmutableList.copyOf(futures), true,
-          MoreExecutors.sameThreadExecutor());
+          MoreExecutors.newDirectExecutorService());
```

Comment:
Since this is since Guava 18, maybe we could use reflection to try `newDirectExecutorService` then fall back to `sameThreadExecutor`, then revert the core bump?

Or copy the impl into this plugin, keeping the Apache license header.

Reply:
nice idea, when I was working on it (and plan to get back to it, although others are welcome to take it up) I was trying to get a green bom PCT build and then look at what's required and see where changes like your suggestion could be applied.

Reply:
I have this change ready to push. Is it ok if I push to your branch @timja?

Reply:
absolutely go for it

Reply:
also see https://github.com/jenkinsci/workflow-api-plugin/pull/135 and https://github.com/jenkinsci/bom/pull/423

Reply:
Errr, I don't have push permission here, so I've filed a PR to your forked branch: https://github.com/timja/workflow-support-plugin/pull/1

Reply:
> also see jenkinsci/workflow-api-plugin#135 and jenkinsci/bom#423

I'll have a look. Meanwhile, isn't this PR mergeable as is?

Reply:
looks good to me